src/typescript/Scala.tmLanguage.ts

```diff
@@ -571,6 +574,26 @@ export const scalaTmLanguage: TmLanguage = {
         }
       }
     },
+    inline: {
+      patterns: [
+        {
+          match: `\\b(inline)\\b(?=(\\s+(?:${plainid}|${backQuotedId})\\s*:)|(.*(val|def|given)))`,
```

Comment:
I think the `.*` is a little bit dangerous, as it can be overly greedy in some instances, and match as long as there is a val somewhere else on the line. For instance:

```scala
inline if (n == 0) 1 else 2; val x = 2
```

is highlighted as: 

![image](https://user-images.githubusercontent.com/990794/92528735-8bb99e00-f229-11ea-85aa-c2e34c2b9f8c.png)


Normally, the `inline` should be highlighted in the same color as the `if`. I think `\s+` could work instead of `.*` (and we could then remove the `\b` as we're matching on `\s+` anyway).

Reply:
I rewrote it and added tests for this case